File: eDentist/eDentist.Services/Database/TransakcijaZaUslugu.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace eDentist.Services.Database
{
    public class TransakcijaZaUslugu
    {
        [Key]
        public int TranskcijaId { get; set; }

        [ForeignKey("DentalServiceId")]
        public int? DentalServiceId { get; set; }
        public DentalService? DentalService { get; set; }

        public double? Amount { get; set; }

        public string? TransactionStatus { get; set; }

        public string? TransId { get; set; }

        [ForeignKey("UserId")]
        public User? User { get; set; }
        public int? UserId { get; set; }

        public DateTime? Datum { get; set; }

        public string? SvrhaUplate { get; set; }
    }
}

File: eDentist/UI/edentist_mobile/lib/models/transakcijaZaUslugu.dart

import 'package:json_annotation/json_annotation.dart';

part "transakcijaZaUslugu.g.dart";

@JsonSerializable()
class TransakcijaZaUslugu {
  int? transkcijaId;
  int? dentalServiceId;
  int? amount;
  String? transactionStatus;
  String? transId;
  int? userId;
  DateTime? datum;
  String? svrhaUplate;
  TransakcijaZaUslugu(this.transkcijaId, this.dentalServiceId, this.amount,
      this.transactionStatus, this.transId, this.userId, this.datum,this.svrhaUplate);

  factory TransakcijaZaUslugu.fromJson(Map<String, dynamic> json) =>
      _$TransakcijaZaUsluguFromJson(json);
  Map<String, dynamic> toJson() => _$TransakcijaZaUsluguToJson(this);
}
